managedRepository/main.tf

```diff
@@ -17,7 +17,7 @@ resource "github_repository" "managed_repository" {
   topics             = "${var.topics}"
 }
 
-resource "github_branch_protection" "managed_repository-master_branch_protection" {
```

Comment:
@dbolack we definitely need a better documentation about the conventions. One other implicit convention we developed while getting to use terraform is that resource names would use the "{}-{}" format, which is exactly what you did. Can you revert the change? 

To summarize the convention:
- we use snake_case for variable names, directory names
- compounded resource names use a `-` as delimiter

Reply:
> @dbolack we definitely need a better documentation about the conventions. One other implicit convention we developed while getting to use terraform is that resource names would use the "{}-{}" format, which is exactly what you did. Can you revert the change?
> 
> To summarize the convention:
> 
> * we use snake_case for variable names, directory names
> * compounded resource names use a `-` as delimiter

I have reverted and added this to the style guide section.